src/main/java/com/solace/samples/java/patterns/GuaranteedProcessor.java

```diff
@@ -147,17 +148,20 @@ public static void main(String... args) throws InterruptedException, IOException
             if (inboundTopic.contains("/pers/pub/")) {  // simple validation of topic
                 // how to "process" the incoming message? maybe do a DB lookup? add some additional properties? or change the payload?
                 OutboundMessageBuilder messageBuilder = messagingService.messageBuilder();
-                messageBuilder.withApplicationMessageId(inboundMsg.getApplicationMessageId());  // set the new message ID to the same as this one
+                if (!StringUtil.isEmpty(inboundMsg.getApplicationMessageId())) {
+                    messageBuilder.withProperty(SolaceProperties.MessageProperties.APPLICATION_MESSAGE_ID, inboundMsg.getApplicationMessageId());// set the new message ID to the same as this one
+                }
                 final String upperCaseTopic = inboundTopic.toUpperCase();  // as a silly example of "processing"
                 OutboundMessage outboundMsg = messageBuilder.build(upperCaseTopic);
-                String [] inboundTopicLevels = inboundTopic.split("/",6);
+                String[] inboundTopicLevels = inboundTopic.split("/", 6);
                 String onwardsTopic = new StringBuilder(TOPIC_PREFIX).append(API.toLowerCase())
-                		.append("/pers/upper/").append(inboundTopicLevels[5]).toString();
+                        .append("/pers/upper/").append(inboundTopicLevels[5]).toString();
```

Comment:
if there is no 5th level we currently throw an exception `java.lang.ArrayIndexOutOfBoundsException: Index 5 out of bounds for length 5` and then no more messages are received. 

We should catch the exception and move on to the next message.

Reply:
Fixed it, have a small try-catch around it with the proper exception handling. Works as expected, logs the error on that message and the continues to the next one.